crates/swc_ecma_lints/src/rules/radix.rs

```diff
@@ -0,0 +1,322 @@
+        ))),
+    }
+}
+
+#[derive(Debug, Default)]
+struct Radix {
+    expected_reaction: LintRuleReaction,
+    top_level_ctxt: SyntaxContext,
+    top_level_declared_vars: AHashSet<Id>,
+
+    radix_mode: RadixMode,
+    check_parens: bool,
+
+    inside_callee: bool,
+    classes_depth: usize,
+    objects_depth: usize,
+    arrow_fns_depth: usize,
+    obj: Option<JsWord>,
+    prop: Option<JsWord>,
+}
+
+impl Radix {
+    fn new(
+        top_level_declared_vars: AHashSet<Id>,
+        top_level_ctxt: SyntaxContext,
+        config: &RuleConfig<RadixConfig>,
+    ) -> Self {
+        let rule_config = config.get_rule_config();
+
+        Self {
+            expected_reaction: config.get_rule_reaction(),
+            top_level_ctxt,
+            top_level_declared_vars,
+
+            radix_mode: rule_config.mode.unwrap_or_default(),
+            check_parens: rule_config.check_parens.unwrap_or(true),
+
+            inside_callee: false,
+            classes_depth: 0,
+            objects_depth: 0,
+            arrow_fns_depth: 0,
+            obj: None,
+            prop: None,
+        }
+    }
+
+    fn emit_report(&self, span: Span, message: &str, note: Option<&str>) {
+        HANDLER.with(|handler| {
+            let mut builder: Option<DiagnosticBuilder> = None;
+
+            match self.expected_reaction {
+                LintRuleReaction::Error => {
+                    builder = Some(handler.struct_span_err(span, message));
+                }
+                LintRuleReaction::Warning => {
+                    builder = Some(handler.struct_span_warn(span, message));
+                }
+                _ => {}
+            };
+
+            if let Some(mut builder) = builder {
+                if let Some(note) = note {
+                    builder.span_note(span, note);
+                }
+
+                builder.emit();
+            }
+        });
+    }
+
+    fn is_inside_class(&self) -> bool {
+        self.classes_depth > 0
+    }
+
+    fn is_inside_object(&self) -> bool {
+        self.objects_depth > 0
+    }
+
+    fn is_inside_arrow_fn(&self) -> bool {
+        self.arrow_fns_depth > 0
+    }
+
+    fn check(&self, call_span: Span, call_expr: &CallExpr) {
+        if let Some(obj) = &self.obj {
+            let obj: &str = &*obj;
+
+            if !OBJ_NAMES.contains(&obj) {
+                return;
+            }
+        }
+
+        let prop = self.prop.as_ref().unwrap();
+
+        if &*prop != "parseInt" {
+            return;
+        }
+
+        if call_expr.args.get(0).is_none() {
+            self.emit_report(call_span, MISSING_PARAMS_MESSAGE, None);
+
+            return;
+        }
+
+        match call_expr.args.get(1) {
+            Some(ExprOrSpread { expr, .. }) => {
+                match &extract_arg_val(expr.as_ref(), self.check_parens) {
+                    ArgValue::Ident => {}
+                    ArgValue::Number(radix) => {
+                        if !(2f64..=36f64).contains(radix) {
```

Comment:
it seems you skip checking for integers; as far as i know `radix` could be equal to `2.5f64` and still accepted.